dali/util/shared_mem.h

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_UTIL_SHARED_MEM_H_
+#define DALI_UTIL_SHARED_MEM_H_
+
+#include <stdint.h>
+#include <memory>
+#include "dali/core/common.h"
+
+namespace dali {
+namespace python {
+
+/**
+ * @brief Creates or simply stores (if provided) file descriptor used for identifying
+ * shared memory chunk. Closes file descriptor in the desctructor if it
+ * wasn't closed earlier.
+ */
+class DLL_PUBLIC ShmFdWrapper {
+ public:
+  /**
+   * @brief Wrap a file descriptor, fd shouldn't be -1.
+   */
+  explicit ShmFdWrapper(int fd);
+
+  /**
+   * Create new file descriptor that can be used for shared memory chunk.
+   */
+  ShmFdWrapper();
+
+  ~ShmFdWrapper();
+
+  DISABLE_COPY_MOVE_ASSIGN(ShmFdWrapper);
+
+  int get_fd();
+
+  int close();
+
+ private:
+  int fd_;
+};
```

Comment:
1. Move this to a separate file (in core?).
2. This class is in no way tied to shared memory - it's just a file handle wrapper, so it should be named so
3. We have a class that manages handles (including move semantics) - UniqueHandle - let's use it.

```suggestion
#if defined(__unix__)
using file_handle_t = int;
#else
#error Platform not supported
#endif

class DLL_PUBLIC FileHandle : public UniqueHandle<file_handle_t, FileHandle> {
 public:
  static void DestroyHandle(file_handle_t h);  // in __unix__ implemented as  { close(h); }
  DALI_INHERIT_UNIQUE_HANDLE(file_handle_t, FileHandle);
};
```

Reply:
Done